community/values/src/main/java/org/neo4j/values/storable/StringValue.java

```diff
@@ -56,6 +56,39 @@ public boolean equals( String x )
     }
 
     @Override
+    public TextValue toLower()
+    {
+        return new StringWrappingStringValue( value().toLowerCase() );
+    }
+
+    @Override
+    public TextValue toUpper()
+    {
+        return new StringWrappingStringValue( value().toUpperCase() );
+    }
+
+    @Override
+    public TextArray split( String separator )
+    {
+        String asString = value();
+        //Cypher has different semantics for the case where the separator
+        //is exactly the value, in cypher we expect two empty arrays
+        //where as java returns an empty array
```

Comment:
Thank you for adding this comment, that is a quite surprising behavior otherwise.